spec/mem/tree_borrows/memory.md

```diff
@@ -193,12 +218,17 @@ impl<T: Target> Memory for TreeBorrowsMemory<T> {
         allocation.deallocation_check(ptr.addr, kind, size, align)?;
 
         // Check that ptr has the permission to write the entire allocation.
-        allocation.extra.root.access(Some(path), AccessKind::Write, Size::ZERO, size)?;
+        allocation.extra.root.access(Some(path), AccessKind::Write, Offset::ZERO, size)?;
+
+        // Check that allocation is not strongly protected.
+        if allocation.extra.root.contains_strong_protector() {
```

Comment:
So this will make it UB to deallocate memory even if there is a zero-sized strongly protected reference, right? Does that match Miri / Coq?
Cc @JoJoDeveloping

Reply:
It matches Coq, idk about miri (but @sun-jacobi can test that himself).

Reply:
It also matches Miri :)

Reply:
In Miri this does not seem to be considered UB, at least under Stacked Borrows:
```rust
use std::alloc::{alloc, dealloc, Layout};

fn test(_x: &mut (), ptr: *mut u8, l: Layout) {
    unsafe { dealloc(ptr, l) };
}

fn main() {
    let l = Layout::from_size_align(1, 1).unwrap();
    let ptr = unsafe { alloc(l) };
    unsafe {test(&mut *ptr.cast::<()>(), ptr, l);}
}
```

Reply:
But it *is* UB under Tree Borrows...

Reply:
Okay so let's match Miri TB for now, though I think we may want to change this in the future.